src/meta/src/hummock/compaction/mod.rs

```diff
@@ -234,36 +234,42 @@ impl Default for ManualCompactionOption {
 
 #[derive(Default)]
 pub struct LocalSelectorStatistic {
-    skip_picker: Vec<(usize, usize, LocalPickerStatistic)>,
+    skip_picker: Vec<LocalPickerStatistic>,
 }
 
 impl LocalSelectorStatistic {
     pub fn report_to_metrics(&self, group_id: u64, metrics: &MetaMetrics) {
-        for (start_level, target_level, stats) in &self.skip_picker {
-            let level_label = format!("cg{}-{}-to-{}", group_id, start_level, target_level);
-            if stats.skip_by_write_amp_limit > 0 {
+        for stats in &self.skip_picker {
+            let level_label = format!("cg{} {} ", group_id, stats.task_label);
+            if stats.skip_by_write_amp_limit() {
                 metrics
                     .compact_skip_frequency
                     .with_label_values(&[level_label.as_str(), "write-amp"])
-                    .inc_by(stats.skip_by_write_amp_limit);
+                    .inc();
             }
-            if stats.skip_by_count_limit > 0 {
+            if stats.skip_by_count_limit() {
                 metrics
                     .compact_skip_frequency
                     .with_label_values(&[level_label.as_str(), "count"])
-                    .inc_by(stats.skip_by_count_limit);
+                    .inc();
             }
-            if stats.skip_by_pending_files > 0 {
+            if stats.skip_by_pending_files() {
                 metrics
                     .compact_skip_frequency
                     .with_label_values(&[level_label.as_str(), "pending-files"])
-                    .inc_by(stats.skip_by_pending_files);
+                    .inc();
             }
-            if stats.skip_by_overlapping > 0 {
+            if stats.skip_by_overlapping() {
                 metrics
                     .compact_skip_frequency
                     .with_label_values(&[level_label.as_str(), "overlapping"])
-                    .inc_by(stats.skip_by_overlapping);
+                    .inc();
+            }
+            if stats.skip_by_trivial_move() {
+                metrics
+                    .compact_skip_frequency
+                    .with_label_values(&[level_label.as_str(), "trivial-move"])
+                    .inc();
             }
```

Comment:
nits: we can unify these codes if we store the skip type (as a string or a enum) in picker stat